core/sql/parser/sqlparser.y

```diff
@@ -500,6 +501,7 @@ static void enableMakeQuotedStringISO88591Mechanism()
 %token <tokval> TOK_CARDINALITY
 %token <tokval> TOK_CASE
 %token <tokval> TOK_CAST
+%token <tokval> TOK_CENTURY
```

Comment:
The new tokens, TOK_CENTURY, TOK_DECADE, etc., need to be added to the nonreserved_word production, otherwise they will be taken as reserved words (notwithstanding the NONRESTOKEN_ flag in ParKeyWords.cpp). Please also add them to test core/TEST037, which tests that non-reserved words really are non-reserved.

Reply:
add TOK_CENTURY, TOK_DECADE, TOK_DOW,  TOK_DOY ,TOK_EPOCH, TOK_WOM to nonreserved_word 
TOK_WEEK, TOK_QUARTER already exists in nonreserved_func_word